CMakeLists.txt

```diff
@@ -1096,7 +1098,7 @@ endif()
 # Check for some functions that are used
 if(WIN32)
   set(CMAKE_REQUIRED_LIBRARIES ws2_32)
-elseif(HAVE_LIBSOCKET)
+elseif(HAVE_LIBSOCKET AND NOT APPLE)
```

Comment:
On apple, surely `HAVE_LIBSOCKET` would never be set so why add an extra variable to check here?

Reply:
it failed to compile without this

Reply:
in what way did it fail to compile, because you are the first to report this and it has been there for a decade